docs/guide/file_management/file_management.md

```diff
@@ -0,0 +1,151 @@
+---
+description: Configurations and management of binary files.
+---
+
+# File management
+
+## Access binary files
+
+To access binary files from the PHP API, use the `Ibexa\Core\IO\IOServiceInterface::loadBinaryFile()` method:
+
+```php
+use Ibexa\Core\IO\IOServiceInterface;
+
+class FileController
+{
+    //...
+
+    $file = $this->ioService->loadBinaryFile($field->value->id);
+    $fileContent = $this->ioService->getFileContents($file);
+    
+    // ...
+}
+```
+
+## Handling binary files
+
+[[= product_name =]] supports multiple binary file handling mechanisms by means of an `IOHandler` interface. This feature is used by the [BinaryFile](../../api/field_types_reference/binaryfilefield.md), [Media](../../api/field_types_reference/mediafield.md) and [Image](../../api/field_types_reference/imagefield.md) Field Types.
+
+### Native IO handler
+
+The IO API is organized around two types of handlers, both used by the IOService:
+
+- `Ibexa\Core\IO\IOMetadataHandler`: stores and reads metadata (such as validity or size)
+- `Ibexa\Core\IO\IOBinarydataHandler`: stores and reads the actual binary data
+
+You can configure IO handlers using semantic configuration. IO handlers are configurable per SiteAccess.
+See the default configuration:
+
+``` yaml
+ibexa:
+    system:
+        default:
+            io:
+                metadata_handler: dfs
+                binarydata_handler: nfs
+The adapter is the *driver* used by Flysystem to read/write files. Adapters are declared using `oneup_flysystem`. 
+Metadata and binary data handlers are configured under `ibexa_io`. See below the configuration for the default handlers. It declares a metadata handler and a binary data handler, both labeled `default`. Both handlers are of type `flysystem`, and use the same Flysystem adapter, labeled `default` as well.
+
+``` yaml
```

Comment:
Previous code block isn't closed. Fixed in 666bdcb4e7eca203de1754c5c8ee81553b699d5e